core/src/main/java/org/pentaho/di/core/DBCacheEntry.java

```diff
@@ -0,0 +1,129 @@
+/*! ******************************************************************************
+ *
+ * Pentaho Data Integration
+ *
+ * Copyright (C) 2002-2017 by Hitachi Vantara : http://www.pentaho.com
+ *
+ *******************************************************************************
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+package org.pentaho.di.core;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.EOFException;
+
+import org.pentaho.di.core.exception.KettleEOFException;
+import org.pentaho.di.core.exception.KettleFileException;
+
+/**
+ * This class represents a single entry in a database cache. A single entry in this case usually means: a single SQL
+ * query.
+ *
+ * @author Matt
+ * @since 15-01-04
+ *
+ */
+public class DBCacheEntry {
+  private String dbname;
+  private String sql;
+  private int hashCode;
+
+  public DBCacheEntry( String dbname, String sql ) {
+    this.dbname = dbname;
+    this.sql = sql;
+  }
+
+  public DBCacheEntry() {
+    this( null, null );
+  }
+
+  public boolean sameDB( String otherDb ) {
+    // ESCA-JAVA0071:
+    if ( dbname == otherDb ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.